src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -94,19 +94,23 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                var testExecution = RunTestOnSeparateThread(context);
-                if (Task.WaitAny(new Task[] { testExecution }, _timeout) != -1
-                    || _debugger.IsAttached)
+                var separateContext = new TestExecutionContext(context)
```

Comment:
I think we need to use `IsolatedContext` which would already to the `CopyOutput` part.
But that relies on setting the `TestExecutionContect.CurrentContext`. 
How does that work with the _context_ parameter.

Reply:
Actually no matter what you do here will help. The test will use whatever the current context is at the time, which might be the context of the next test running.

So we can only solve the case where the test doesn't time out.
So either your fix or not creating a separate context at all.